feat: [SEC-7263] Add dependency-scan GitHub Actions workflow

pkaeding opened the pull request:
## Summary

Adds a GitHub Actions workflow to generate Software Bill of Materials (SBOM) for Node.js dependencies and evaluate them against security policies as part of SEC-7263.

**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests) 
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Security ticket: SEC-7263

**Describe the solution you've provided**

This PR adds `.github/workflows/dependency-scan.yml` which:

- **Generates SBOM**: Uses `launchdarkly/gh-actions/actions/dependency-scan/generate-sbom@main` to create a Software Bill of Materials for Node.js dependencies
- **Evaluates policies**: Uses `launchdarkly/gh-actions/actions/dependency-scan/evaluate-policy@main` to check generated SBOMs against security policies
- **Triggers appropriately**: Runs on pull requests and pushes to main branch
- **Job dependency**: `evaluate-policy` job waits for `generate-nodejs-sbom` to complete

**Key Review Points**

⚠️ **Critical items to verify:**
1. **Workflow syntax** - YAML structure and GitHub Actions syntax
2. **Action availability** - Confirm `launchdarkly/gh-actions` actions are accessible for this public repository
3. **Artifact handling** - Verify the `bom-*` pattern correctly matches artifacts generated by the first job
4. **Permissions** - Check if additional permissions are needed for artifact access or security scanning
5. **Branch configuration** - Confirm `main` is the correct default branch for this repository

**Additional context**

This workflow follows the established pattern for LaunchDarkly public repositories using `launchdarkly/gh-actions` (rather than `launchdarkly/common-actions` used for private repos). The workflow has not been tested locally, so initial CI run will validate configuration.

Link to Devin run: https://app.devin.ai/sessions/434bb14b7bac4d81b9979b88965be92b  
Requested by: @pkaeding

devin-ai-integration[bot] committed:
feat: [SEC-7263] Add dependency-scan GitHub Actions workflow

Generate Node.js SBOM using launchdarkly/gh-actions for SEC-7263.
Add policy evaluation step with bom-* artifacts pattern.
Configure triggers for pull requests and main branch pushes.

Co-Authored-By: Patrick Kaeding <patrick@kaeding.name>

kinyoklion commented on .github/workflows/dependency-scan.yml:
```diff
@@ -0,0 +1,30 @@
+  generate-nodejs-sbom:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
Don't we use pinned SHAs for these?

pkaeding replied:
Devin: run the `pin-github-action` tool to get the proper sha

devin-ai-integration[bot] committed:
fix: use pinned SHA for actions/checkout instead of version tag

Address GitHub comment requesting pinned commit SHAs for security.
Replace actions/checkout@v4 with pinned SHA 692973e3d937129bcbf40652eb9f2f61becf3332.

Co-Authored-By: Patrick Kaeding <patrick@kaeding.name>

kinyoklion commented on .github/workflows/dependency-scan.yml:
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
```
I think the latest 4 is 08eba0b27e820071cde6df949e0beb9ba4906955?

kinyoklion reviewed (changes requested):
Need to get the right SHA. Not sure which one it chose.

devin-ai-integration[bot] committed:
fix: use correct pinned SHA for actions/checkout@v4

Address GitHub comment from kinyoklion requesting correct SHA.
Update to use 08eba0b27e820071cde6df949e0beb9ba4906955 instead of
692973e3d937129bcbf40652eb9f2f61becf3332.

Co-Authored-By: Patrick Kaeding <patrick@kaeding.name>
